### PR TITLE
1027 sync v43 to v44

### PR DESCRIPTION
**Pull request by zmstone**
## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/` dir
- [x] For EMQX 4.x: `appup` files updated (execute `scripts/update-appup.sh emqx`)
- [ ] For internal contributor: there is a jira ticket to track this change, and another jira tickt to track doc updates (if any)
- [ ] In case of non-backward compatible changes, reviewer should check this item as a write-off, and add details in **Backward Compatibility** section

### CODE REVIEW

File: apps/emqx_management/src/emqx_mgmt_data_backup.erl [hunk 1]
```diff
@@ -871,8 +872,14 @@ backup_dir_old_version() ->
     emqx:get_env(data_dir).
 
 legal_filename(Filename) ->
-    MaybeJson = filename:extension(Filename),
-    MaybeJson == ".json" orelse MaybeJson == <<".json">>.
+    case to_unicode_bin(Filename) of
+        <<"/", _/binary>> ->
+            %% never allow abs path
+            false;
```
Comment (thalesmg) [inline apps/emqx_management/src/emqx_mgmt_data_backup.erl:878]: Not the scope here, but would probably have to check if the normalized path is in some allow-listed directories?

File: apps/emqx_management/src/emqx_mgmt_api_data.erl [hunk 2]
```diff
@@ -162,7 +162,7 @@ import_content(Content) ->
 tmp_filename() ->
     Seconds = erlang:system_time(second),
     {{Y, M, D}, {H, MM, S}} = emqx_mgmt_util:datetime(Seconds),
-    list_to_binary(io_lib:format("emqx-export-~p-~p-~p-~p-~p-~p.json", [Y, M, D, H, MM, S])).
+    iolist_to_binary(io_lib:format("emqx-export-~p-~p-~p-~p-~p-~p.json", [Y, M, D, H, MM, S])).
```
Comment (thalesmg) [inline apps/emqx_management/src/emqx_mgmt_api_data.erl:165]: This one doesn't need `unicode:characters_to_binary`?

**Review by thalesmg — DISMISSED** [review]

File: apps/emqx_management/src/emqx_mgmt_data_backup.erl (hunk 1 above)
Comment (zmstone) [reply apps/emqx_management/src/emqx_mgmt_data_backup.erl:878]: You are right.
Will revert this change and add another task to follow up

File: apps/emqx_management/src/emqx_mgmt_api_data.erl (hunk 2 above)
Comment (zmstone) [reply apps/emqx_management/src/emqx_mgmt_api_data.erl:165]: This is certainly going to be latin1

**Pull request merged**